lib/plugin/mk-plugin.nix

```diff
@@ -0,0 +1,90 @@
+{
+  lib,
+  nixvimOptions,
+  nixvimUtils,
+}:
+with lib;
+config:
+{
+  name,
+  maintainers,
+  url ? defaultPackage.meta.homepage,
+  imports ? [ ],
+  description ? null,
+  # deprecations
+  optionsRenamedToSettings ? [ ],
+  # colorscheme
+  isColorscheme ? false,
+  colorscheme ? name,
+  # options
+  originalName ? name,
+  defaultPackage ? null,
+  extraOptions ? { },
```

Comment:
I wonder if there's a way we can pass in a (curried, wrapped, or custom) settings-option-factory function? The only difference between nvim and vim's settings is the description, right?

Reply:
If the description is the only real difference, maybe have `mkPlugin` accept a `settingsDesc` argument?